Sources/IO/Core/HttpDataSetReader/index.js

```diff
@@ -247,6 +238,29 @@ function vtkHttpDataSetReader(publicAPI, model) {
     return publicAPI.updateMetadata(!!options.loadData);
   };
 
+  publicAPI.parseObject = (
+    manifest,
+    { loadData, baseUrl, preserveManifest = true }
+  ) => {
+    if (baseUrl) {
+      model.baseURL = baseUrl;
+    }
+
+    const dataset = preserveManifest ? structuredClone(manifest) : manifest;
```

Comment:
processDataset will mutates the input object. Since parseObject is publicly exposed I don't think we should mutates the user provided object by default

Reply:
@jourdain @floryst WDYT ?

Reply:
Fair enough, then maybe return "preserveManifest" into a "deepCopy" or "clone" in order to keep the vocabulary consistent with the rest of VTK.js